src/serac/physics/solid.hpp

```diff
@@ -326,6 +338,19 @@ class Solid : public BasePhysics {
    */
   void advanceTimestep(double& dt) override;
 
+  /**
+   * @brief Solve the adjoint problem
+   * @note It is expected that the forward analysis is complete and the current displacement state is valid
+   * @note If the essential boundary state is not specified, homogeneous essential boundary conditions are applied
+   *
+   * @param[in] adjoint_load_form The linear form that when assembled contains the right hand side of the adjoint system
+   * @param[in] state_with_essential_boundary A optional finite element state containing the non-homogenous essential
+   * boundary condition data for the adjoint problem
+   * @return The computed adjoint finite element state
+   */
+  virtual const serac::FiniteElementState& solveAdjoint(mfem::ParLinearForm& adjoint_load_form,
```

Comment:
Is the `adjoint_load_form` guaranteed to always be `LinearForm` or would it be more versatile possibly as a `NonlinearForm` and then use serac wrappers?

Reply:
I believe the adjoint load will always be a linear form. However, I think the plan (perhaps as soon as the next PR?) is to change the interface to expect a more standard serac type, either a `Functional` or a `HypreParVec`. We used a `ParLinearForm` here since that is what LiDO is using currently. We plan to start wrapping and using serac types soon, though.

Reply:
I also believe adjoint loads will always be linear forms since they are derived from the explicit derivative of the QoI with respect to the state variable. That said, the plan as @sethwatts states is to move this to a more standard `Functional` or `Vector` type.

Reply:
I think `Functional` has notions of being able to get gradient while a plain old `Vector` does not.  I was mainly wonder if that notion was important (similar distinction between `LinearForm` and `NonLinearForm`.